crates/api/src/v1/monitor.rs

```diff
@@ -0,0 +1,124 @@
+//! Types relating to the monitor API.
+
+use serde::{Deserialize, Serialize, Serializer};
+use std::borrow::Cow;
+use thiserror::Error;
+
+/// Represents checkpoint verification response.
+#[derive(Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct CheckpointVerificationResponse {
+    /// The checkpoint verification state.
+    pub checkpoint: CheckpointVerificationState,
+    /// The checkpoint signature verification state.
+    pub signature: CheckpointSignatureVerificationState,
+}
+
+/// Represents checkpoint verification state.
+#[derive(Eq, PartialEq, Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub enum CheckpointVerificationState {
+    /// The checkpoint signature is unverified and could be valid or invalid.
+    #[serde(rename_all = "camelCase")]
+    Unverified,
+    /// The checkpoint is verified.
+    #[serde(rename_all = "camelCase")]
+    Verified,
+    /// The checkpoint log length does not exist.
+    #[serde(rename_all = "camelCase")]
+    NotFound,
```

Comment:
I'm still unclear on the difference between `unverified` and `notFound`. I guess a concrete question to ask is: What would a client do differently if it received one vs the other?

Reply:
`unverified` is when the server is unable to verify whether valid or invalid. This may be terminal. Or, currently, there is a `503` that instructs to retry.

`notFound` is an invalid checkpoint due to the registry either never produced a checkpoint with the given `logLength` (between two other valid ones) or it is definitively in the future.

Reply:
@esoterra was thinking the reason why something failed was important information.

Feels like we need to communicate the minimum options: `retry`, `verified`, `invalid`, `unknown`

Reply:
That's fine, its just that if *I* don't understand the difference it isn't likely that many other people will either. :slightly_smiling_face:

Reply:
Updated, from our discussion